### PR TITLE
Auto-connect improvements for node client

### PR DESCRIPTION
**Pull request by javo**
Auto-connect logic reimplemented for node client.
- When device is disconnected scans the nearests open access points and tries to connect one by one until the 'connected' event is triggered.
- If it can't achieve the connection waits a minute and the process starts again.
- Added `connected_internet` and `disconnected_internet` events so it can be posible to discriminate when the control panel isn't available and the device disconnects to internet.
- Save the `active_access_point` data and only deleted when the device disconnects to the internet.

### CODE REVIEW

**Comment by michelson** [comment]
no specs?

**Comment by javo** [comment]
ya van don @michelson

File: lib/agent/providers/network/index.js [hunk 1]
```diff
@@ -135,7 +135,7 @@ exp.get_open_access_points_list = function(callback) {
       return callback(err || new Error('No access points detected.'));
 
     var open_aps = list.filter(function(ap) {
-      return ap.security === false;
+      return ap.security === false || ap.security == null;
```
Comment (michelson) [inline lib/agent/providers/network/index.js]: ¿porque security lo comparas con === y el otro == ??
creo que si le pones `return !ap.security` igual debería funcionar

File: lib/agent/triggers/auto-connect/index.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  }
+  else restart_reconnection();
+}
+
+var wait_normal_reconnection = function() {
+  logger.info("Device disconnected! Waiting for reconnection...");
+  timer = setTimeout(function() {
+    logger.info("Nothing happened, let's try connecting to the nearest access points...");
+    reconnect.get_open_ap_list(function(err, list) {
+      if (err) check_err(err);
```
Comment (michelson) [inline lib/agent/triggers/auto-connect/index.js]: if else innecesario, creo

**Comment by michelson** [comment]
@javo correcciones muy menores no mas. eso si, sería bacan correr los specs en travis.ci

File: lib/agent/triggers/auto-connect/linux.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    exec(cmd, function(err, stdout) {
+      cb(null, stdout.split("\n").slice(0, -1));
+    })
+  }
+  net_interface ? get_profiles() : get_interface(get_profiles);
+}
+
+exports.create_profile = function(ssid, cb) {
+  var create = function() {
+    var cmd = 'nmcli connection add type wifi ifname ' + '"' +  net_interface + '"' + ' con-name '  + '"' +  ssid  + '"' +  ' ssid '  + '"' +  ssid + '"';
```
[the remark this answers is not in the document]
Comment (michelson) [reply lib/agent/triggers/auto-connect/linux.js]: 👍

**Pull request merged**